core/src/main/java/io/parsingdata/metal/token/IterableToken.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2013-2018 Netherlands Forensic Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.parsingdata.metal.token;
+
+import io.parsingdata.metal.Trampoline;
+import io.parsingdata.metal.data.Environment;
+import io.parsingdata.metal.data.ParseState;
+import io.parsingdata.metal.encoding.Encoding;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static io.parsingdata.metal.Trampoline.complete;
+import static io.parsingdata.metal.Trampoline.intermediate;
+import static io.parsingdata.metal.Util.checkNotNull;
+import static io.parsingdata.metal.Util.success;
+
+public abstract class IterableToken extends Token {
+
+    public final Token token;
+
+    IterableToken(String name, final Token token, Encoding encoding) {
+        super(name, encoding);
+        this.token = checkNotNull(token, "token");
+    }
+
+    protected final Optional<ParseState> parse(final Environment environment, final Function<Environment, Boolean> stopCondition, final Function<Environment, Optional<ParseState>> ifIterationFails) {
```

Comment:
`Function<Env, Boolean>` = `Predicate<Env>` :) (unless you would want it to return a null)

Reply:
Tnx!